plugin/pkg/scheduler/scheduler.go

```diff
@@ -90,39 +90,40 @@ func (s *Scheduler) scheduleOne() {
 	dest, err := s.config.Algorithm.Schedule(pod, s.config.NodeLister)
 	if err != nil {
 		glog.V(1).Infof("Failed to schedule: %+v", pod)
-		s.config.Recorder.Eventf(pod, api.EventTypeWarning, "FailedScheduling", "%v", err)
 		s.config.Error(pod, err)
+		s.config.Recorder.Eventf(pod, api.EventTypeWarning, "FailedScheduling", "%v", err)
 		return
 	}
 	metrics.SchedulingAlgorithmLatency.Observe(metrics.SinceInMicroseconds(start))
 
-	b := &api.Binding{
-		ObjectMeta: api.ObjectMeta{Namespace: pod.Namespace, Name: pod.Name},
-		Target: api.ObjectReference{
-			Kind: "Node",
-			Name: dest,
-		},
-	}
+	// Assume optimistically that the binding will succeed and send it
+	// in the background.
+	// The internal cache only assumes a pod anyway and waits for it to be
```

Comment:
Please make this sentence a little clearer about three things:
* "the internal cache only assumes a pod anyway": does this mean "the internal cache only assumes a pod, and does not write it back to the API server until (when?)" ?
* "waits for it to be read via watch": waits for it to be read via watch before doing what?
* "so that shouldn't cause problems": what kind of problem does it not cause? (this is related to the first point -- what kind of problem would there be if it were not the case that "the internal cache only assumes a pod"?)